lib/event.c

```diff
@@ -993,6 +997,23 @@ void _event_add_read_write(const struct xref_eventsched *xref,
 #endif
 				break;
 			}
+			/*
+			 * We are setting the fd = -1 for the
+			 * case when a read/write event is going
+			 * away.  if we find a -1 we can stuff it
+			 * into that spot, so note it
+			 */
+			if (m->handler.pfds[i].fd == -1 && earlierqueuepos == m->handler.pfdcount)
+				earlierqueuepos = i;
```

Comment:
You can just check and set `queuepos` here and remove `earlierqueuepos` and `queuepos_found`… since the loop is gonna continue to run, and if we find the fd we overwrite `queuepos` anyway and `break` after…

Reply:
sure